app/assets/javascripts/admin/addon/components/admin-page-header.gjs

```diff
@@ -41,14 +51,75 @@ export default class AdminPageHeader extends Component {
           <h1 class="admin-page-header__title">{{this.title}}</h1>
         {{/if}}
 
-        <div class="admin-page-header__actions">
-          {{yield
-            (hash
-              Primary=PrimaryButton Default=DefaultButton Danger=DangerButton
-            )
-            to="actions"
-          }}
-        </div>
+        {{#if (has-block "actions")}}
+          <div class="admin-page-header__actions">
+            {{#if this.site.mobileView}}
+              <DMenu
+                @identifier="admin-page-header-mobile-actions"
+                @title={{i18n "more_options"}}
+                @icon="ellipsis-vertical"
+                class="btn-small"
+              >
+                <:content>
+                  <DropdownMenu class="admin-page-header__mobile-actions">
+                    {{yield
+                      (hash
+                        Primary=PrimaryActionListItem
+                        Default=DefaultActionListItem
+                        Danger=DangerActionListItem
+                        Wrapped=WrappedActionListItem
+                      )
+                      to="actions"
+                    }}
+                  </DropdownMenu>
+                </:content>
+              </DMenu>
+            {{else}}
+              {{yield
+                (hash
+                  Primary=PrimaryButton
+                  Default=DefaultButton
+                  Danger=DangerButton
+                  Wrapped=WrappedButton
+                )
+                to="actions"
+              }}
+            {{/if}}
+          </div>
+        {{else if @headerActionComponent}}
+          <div class="admin-page-header__actions">
+            {{#if this.site.mobileView}}
```

Comment:
It's annoying that I have to repeat essentially the same code as above except using `@headerActionComponent` instead of `yield`, not sure how to get around that though

Reply:
Moving `{{#if (has-block "actions")}}` check deeper (inside `<DropdownMenu>` and in the `else` branch) could reduce the repetition/number of lines slightly?

Reply:
But then we would end up rendering these empty components, no?

Reply:
yeah, we'd need an outer check still, but we can save some lines anyway